lucene/facet/src/java/org/apache/lucene/facet/sortedset/ConcurrentSortedSetDocValuesFacetCounts.java

```
@@ -159,13 +160,15 @@ private FacetResult getDim(String dim, OrdRange ordRange, int topN) throws IOExc
     final MatchingDocs hits;
     final OrdinalMap ordinalMap;
     final int segOrd;
+    final Bits liveDocs;
 
     public CountOneSegment(
         LeafReader leafReader, MatchingDocs hits, OrdinalMap ordinalMap, int segOrd) {
       this.leafReader = leafReader;
       this.hits = hits;
       this.ordinalMap = ordinalMap;
       this.segOrd = segOrd;
+      this.liveDocs = (leafReader != null) ? leafReader.getLiveDocs() : null;
```

I wonder if it makes sense to just `assert leafReader != null` at the top of this ctor? If it's `null`, when `call()` gets invoked it will throw an NPE anyway (down in `DocValues.getSortedSet`), so might be nice to fail early on this. What do you think?

---

Yes that makes sense. I made the change in the new revision captured in a different PR
https://github.com/apache/lucene/pull/293